### DISCUSSION TITLE
Username from Github as Author

### DISCUSSION THREAD

**Discussion by Larissartoricader** [neuefische/questions#840] [Web] [answered]
For our [Herbie App,](https://capstone-git-styling-homep-397f67-larissasartoricaders-projects.vercel.app/) we want to display the GitHub username as the author name under each user's recipe. Currently, we can display the user email, which we use to allow logged-in users to edit or delete only their own recipes. Therefore, the user email is already part of the object in our backend. I attempted to do the same with the username, saving it in our backend and displaying it as the author, but unfortunately, it does not work with the username.

### api/recipe/index.js

```
if (request.method === "POST") {
    try {
      const recipeData = request.body;
      const owner = session.user.email; //It works
      const author = session.user.username; //It does not work

      const imageUrl = await generateImage(request.body.title);
      
      await Recipe.create({
        ...recipeData,
        image: imageUrl,
        owner, //It works
        author, //It does not work
      });

      response.status(201).json({ status: "Recipe created." });
      return;
    } catch (error) {
      response.status(400).json({ error: error.message });
      return;
    }
```
    
  ###  I also updated the Schema:
    

```
 const RecipeSchema = new Schema({
  title: { type: String, required: true },
  ingredients: [{ type: String, required: true }],
  preparation: { type: String, required: true },
  usage: { type: String, required: true },
  symptoms: [{ type: String, required: true }],
  image: { type: String },
  owner: { type: String, required: true },
  author: { type: String, required: false },
});
```

I wonder if there is some kind of limitation to saving the user name like we have done with the e-mail. 

 ### MongoDB
<img width="743" alt="Screenshot 2024-05-22 at 17 07 33" src="https://github.com/neuefische/questions/assets/152177654/2bfea21a-6ab3-44f4-be30-fc16a031d958">
At moment I can only see the owner with the user-email, but no author with the username.

**Comment by Jessica-Windoffer** [comment]
Have you tried using `const author = session.user.**name**;` instead of `const author = session.user.**username**`? I think that is how we did it and it worked 😊

**Reply by Jessica-Windoffer** [reply, marked as answer]
That how it looks in our Code - this way you can also access profile pictures from Google/Github


```js
import { useSession } from "next-auth/react";

const { data: session } = useSession();

<LoginDataContainer>
   <ProfileImage
      src={session.user.image}
      width={"100"}
      height={"100"}
      alt="user-image"
   />
     <UserGreeting>Hello, {session.user.name}!</UserGreeting>
</LoginDataContainer>
```
